### PR TITLE
SKU sdk - update credential matching logic

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/brave-browser/issues/32924

Test plan TBD

Minimal version of https://github.com/brave/brave-core/pull/20139

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/skus/browser/rs/lib/src/sdk/credentials/mod.rs [hunk 1]
```diff
@@ -49,9 +49,10 @@ where
                     if let Some(expires_at) = expires_at {
                         // attempt to refresh credentials if we're within 5 days of expiry
                         if Utc::now().naive_utc() > (expires_at - chrono::Duration::days(5)) {
+                            error!("Within 5 days of expiry; refreshing order credentials.");
                             let refreshed = self.refresh_order_credentials(order_id).await;
                             if refreshed.is_err() {
-                                continue;
+                                error!("Error refreshing order credentials.");
                             }
                         }
```
Comment (antonok-edm) [inline components/skus/browser/rs/lib/src/sdk/credentials/mod.rs:57]: looks like we have this identical code in 2 locations (just highlighting for later)

**Review by antonok-edm — COMMENTED** [review]
rust code seems alright to me

**Review by antonok-edm — APPROVED** [review]
sorry meant to approve originally :sweat_smile:

**Comment by bsclifton** [comment]
Requires `1.60.30` or later to test

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

```
Brave | 1.60.38 Chromium: 117.0.5938.88 (Official Build) nightly (64-bit)
-- | --
Revision | 89f7ea7a01a3ab15163916a9441c473ab1021372
OS | Windows 10 Version 22H2 (Build 19045.3448)
```

### Ensure new purchases work - PENDING
<details>

Fresh profile
Login to account.brave.com with a new account
Go through steps to purchase VPN
Wait for the YOU HAVE ACTIVE CREDENTIALS LOADED banner over VPN
Verify that you can connect/disconnect from servers

</details>

### Existing users (recent purchase) - PENDING
<details>

Fresh profile
Login to account.brave.com with an account that already has VPN
Wait for the YOU HAVE ACTIVE CREDENTIALS LOADED banner over VPN
Verify that you can connect/disconnect from servers

</details>

### Existing users - within 5 days of renewal - PENDING
<details>

Note: this can be tricky to configure, since it requires the backend and client be in sync (ex: you can't change system clock).
Help may be needed to locate an account (ex: staging account using mailinator) that is within 5 days of the expires_at (in brave://skus-internals)

Fresh profile
Login to account.brave.com with an account that already has VPN
Wait for the YOU HAVE ACTIVE CREDENTIALS LOADED banner over VPN
Verify that you can connect/disconnect from servers
Visit brave://skus-internals
click Fetch SKUs state
Collapse the big creds section and look under that for an array of unblinded_creds:
There should be only the number you're expecting, until the expiration date occurs. The person needs to login again after that.

</details>

**Pull request merged**